AK/URL.cpp

```diff
@@ -519,6 +519,127 @@ bool URL::is_special_scheme(const String& scheme)
     return scheme == "ftp" || scheme == "file" || scheme == "http" || scheme == "https" || scheme == "ws" || scheme == "wss";
 }
 
+String URL::serialize_data_url() const
+{
+    VERIFY(m_scheme == "data");
+    VERIFY(!m_data_mime_type.is_null());
+    VERIFY(!m_data_payload.is_null());
+    StringBuilder builder;
+    builder.append(m_scheme);
+    builder.append(':');
+    builder.append(m_data_mime_type);
+    if (m_data_payload_is_base64) {
+        builder.append(";base64");
+    }
+    builder.append(',');
+    builder.append(m_data_payload);
```

Comment:
I wasn't sure. Since this is the last field in the data URL, there are no more control characters. Perhaps this is a case of "better safe than sorry", and we should at least encode this with the `C0Control` set, so all control and non-ASCII characters are filtered out.